File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyMemberRemoveMessage.scala





















// Generated on 05/03/2013 10:34:47
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyMemberRemoveMessage extends AbstractPartyEventMessage {
  override val messageId = 5579

  protected trait Message extends AbstractPartyEventMessage.Message with Serializable {
    override val messageId = PartyMemberRemoveMessage.messageId

    val leavingPlayerId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(leavingPlayerId)
    }

  }

  protected def create(partyId: Int, leavingPlayerId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val leavingPlayerId = buf.readInt()

    create(partyId, leavingPlayerId)
  }

}

object PartyMemberRemoveMessage extends PartyMemberRemoveMessage {
  case class MessageImpl(partyId: Int, leavingPlayerId: Int) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, leavingPlayerId: Int) = MessageImpl(partyId, leavingPlayerId)

  def apply(partyId: Int, leavingPlayerId: Int) = create(partyId, leavingPlayerId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PartyMemberRemoveMessage.java


// Generated on 05/08/2013 19:37:49
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyMemberRemoveMessage extends AbstractPartyEventMessage {
    public static final int MESSAGE_ID = 5579;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int leavingPlayerId;
    
    public PartyMemberRemoveMessage() { }
    
    public PartyMemberRemoveMessage(int partyId, int leavingPlayerId) {
        super(partyId);
        this.leavingPlayerId = leavingPlayerId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(leavingPlayerId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        leavingPlayerId = buf.readInt();
        if (leavingPlayerId < 0)
            throw new RuntimeException("Forbidden value on leavingPlayerId = " + leavingPlayerId + ", it doesn't respect the following condition : leavingPlayerId < 0");
    }
    
}
